pkg/cmd/promhelperservice/handlers/instanceconfigs/helpers.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 The Cockroach Authors.
+
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package instanceconfigs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+)
+
+const (
+	// resourceName and version for the API
+	resourceName    = "instance-configs"
+	resourceVersion = "v1"
+	// ClusterIDParam is the url parameter
+	ClusterIDParam = "cluster_id"
+	// defaultInstanceConfigDir is the default instance config directory from where the yml files are created/removed.
+	defaultInstanceConfigDir = "/opt/prom/prometheus/instance-configs"
+	instanceConfigFileExt    = "yml"
+	// instanceConfigEnvVar needs to be set to override the defaultInstanceConfigDir
+	instanceConfigEnvVar = "INSTANCE_CONFIG_DIR"
+)
+
+// FormInsConfUrlWithClusterID returns the url with cluster ID for instance config
+func FormInsConfUrlWithClusterID() string {
+	return fmt.Sprintf("/%s/%s/{%s}", resourceVersion, resourceName, ClusterIDParam)
+}
+
+// getDir returns the instance config dir. The value can be read from
+// env instanceConfigEnvVar, else the defaultInstanceConfigDir is used.
+func getDir() string {
```

Comment:
Nit: IMO, this is also unnecessary logic. We can force the caller to pass a directory when starting the server (no implicit lookup of env var or code-default).